pkg/objstore/clientutil/parse.go

```diff
@@ -35,7 +35,8 @@ func ParseContentLength(m http.Header) (int64, error) {
 
 // ParseLastModified returns the timestamp parsed from the Last-Modified
 // HTTP header in input (expected to be in the RFC3339 format).
-func ParseLastModified(m http.Header) (time.Time, error) {
+// Passing an optional second parameter to override the default format.
+func ParseLastModified(m http.Header, f ...string) (time.Time, error) {
```

Comment:
I would vote for making this argument explicit. The problem by this way of doing `optional` arguments it's just it's end being quick hacky and surprising. 

It's totally ok to pass everywhere `time.RFC3339` or even allow passing "" and setting to time.RFC3339 if that happens.